src/Mvc/Mvc.ViewFeatures/src/HtmlHelperRazorComponentExtensions.cs

```diff
@@ -22,14 +21,14 @@ public static class HtmlHelperRazorComponentExtensions
         /// </summary>
         /// <param name="htmlHelper">The <see cref="IHtmlHelper"/>.</param>
         /// <returns>The HTML produced by the rendered <typeparamref name="TComponent"/>.</returns>
-        public static Task<IHtmlContent> RenderComponentAsync<TComponent>(this IHtmlHelper htmlHelper) where TComponent : IComponent
+        public static Task<IHtmlContent> RenderStaticComponentAsync<TComponent>(this IHtmlHelper htmlHelper) where TComponent : IComponent
```

Comment:
I know this is the first iteration. At some point we might want to do some API review on this (cc @danroth27 @rynowak and maybe @davidfowl if interested). If I understand correctly, the current state is:

 * `@(await Html.RenderComponentAsync<MyComponent>())` opens a server-side circuit, attaches the component to it, prerenders the component into the output, and keeps the circuit in memory to wait for the subsequent client reconnect, at which point it becomes interactive
 * `@(await Html.RenderStaticComponentAsync<MyComponent>())` just prerenders the component into the output, without creating or retaining any circuit

The other obvious option would be to have `RenderComponentAsync` be the static case, and `RenderInteractiveComponentAsync` be for interactive (so you know you're buying into the extra server-side cost). I'm not saying I think that's necessarily better, just exploring the options. Depends on what we think the mainstream scenarios are.

I'm guessing for this release we don't want to have any way to open a circuit and then do the prerender+interactive thing for multiple top-level components in that circuit, right?

Reply:
I totally agree with you in that we need to revisit the naming. My priority right now is to separate RenderComponentAsync from PrerenderComponentAsync because it's too similar and causes confusion.

> I'm guessing for this release we don't want to have any way to open a circuit and then do the prerender+interactive thing for multiple top-level components in that circuit, right?

This is already supported. We will reuse the circuit when you do multiple `RenderComponentAsync()`.

The only missing piece is mixing prerendered and non prerendered components on the same circuit but that is coming.

Reply:
I think what's here is **good enough** for preview 4 unless there's a suggestion we think is *much better*. Just be aware that people will be drawn to the shorter/less-complicated name. As it stands we're going to end up with people defaulting to the prerendering+circuit case.